core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -792,7 +793,11 @@ private void handleRequestOrResponseComplete() {
                         // Stop receiving new requests.
                         handledLastRequest = true;
                         if (unfinishedRequests.isEmpty()) {
-                            ctx.writeAndFlush(Unpooled.EMPTY_BUFFER).addListener(CLOSE);
+                            ctx.channel().eventLoop().schedule(() -> {
+                                if (ctx.channel().isActive()) {
+                                    ctx.writeAndFlush(Unpooled.EMPTY_BUFFER).addListener(CLOSE);
+                                }
+                            }, Flags.defaultHttp1ConnectionCloseDelayMillis(), TimeUnit.MILLISECONDS);
```

Comment:
We don't need to schedule if `Flags.defaultHttp1ConnectionCloseDelayMillis() == 0`.
```java
final long closeDelay = Flags.defaultHttp1ConnectionCloseDelayMillis();
if (closeDelay == 0) {                                
    ctx.writeAndFlush(Unpooled.EMPTY_BUFFER).addListener(CLOSE);
} else {
   // schedule the channel closing task.
}
```

Reply:
Thanks for good suggestions 🙇 I changed not to shedule when `defaultHttp1ConnectionCloseDelayMillis` is 0 and clarified it to Javadoc.